### PR TITLE
Fix acceptance test bug in IBAN input

### PR DESCRIPTION
**Pull request by JulianRoesner**
# Short Description
Our acceptance test sometimes run into odd behavior and fail because the IBAN number is incorrect (s. this run for example). I had a look into that and ran them locally. After up to 24 runs, they would fail for the same reason. Although the input should be correct, I would receive an incorrect IBAN number in the backend with one digit placed incorrectly.
I assume that this is because of some timing issue with `cy.type()`. `type` simulates real typing and by default has a delay between input of different digits of 10 ms. We use masking for the IBAN input, which changes the input while typing. I assume that this sometimes changes at the wrong time and the digits get misplaced because of that. The misplaced digits were always at that point of the IBAN where we add a space. [This discussion](https://stackoverflow.com/questions/67416126/cypress-insert-wrong-data-from-json) also points in that direction.

To fix this problem I increased the delay for the IBAN input to 50ms (recommendation taken from discussion mentioned above). I ran the acceptance tests locally over 50 times without getting the error again. Therefore, I assume that this should fix it.

# Changes
- Add a delay to IBAN input
- Removed unneccessary tax number tests: We test that already in the component itself. This is still legacy code from before, when we did not have the functional tests.
- Fix MockErica tax_number_validity: It was hardcoded before and now it always takes the last to parameters.

# Feedback
- What to you think about the deleted acceptance tests and the changes to MockErica?

# How to review this Pull Request
[Link to Confluence](https://digitalservice4germany.atlassian.net/wiki/spaces/STL/pages/117637157/Development+Process#Approve-or-comment) (internal)

### CODE REVIEW

File: acceptance_tests/cypress/integration/lotse_flow.spec.js [hunk 1]
```diff
@@ -215,40 +215,6 @@ context('Acceptance tests', () => {
             cy.get('div[id=familienstand_zusammenveranlagung_field]').should('not.be.visible')
         })
 
-        it('Enter different tax number data ', () => {
```
Comment (SannyNguyenHung) [inline acceptance_tests/cypress/integration/lotse_flow.spec.js:218]: Why did you delete this test?

File: webapp/tests/elster_client/mock_erica.py [hunk 2]
```diff
@@ -84,7 +84,7 @@ def mocked_elster_requests(*args, **kwargs):
                 response = MockErica.get_tax_offices()
             elif _PYERIC_API_BASE_URL_01 + '/tax_number_validity' in args[0]:
                 sub_urls = args[0].split('/')
-                response = MockErica.is_valid_tax_number(state_abbreviation=sub_urls[2], tax_number=sub_urls[3])
+                response = MockErica.is_valid_tax_number(state_abbreviation=sub_urls[len(sub_urls)-2], tax_number=sub_urls[len(sub_urls)-1])
```
Comment (SannyNguyenHung) [inline webapp/tests/elster_client/mock_erica.py:87]: Can we get a Length < 0? why did you change that?

**Review by SannyNguyenHung — APPROVED** [review]
LGTM :) just 2 questions

File: webapp/tests/elster_client/mock_erica.py (hunk 2 above)
Comment (JulianRoesner) [reply webapp/tests/elster_client/mock_erica.py:87]: I ran the acceptance tests with the mockEria and got problems. This makes sense because the erica_url is: http://127.0.0.1/v1

Sending a request for the tax number validity would then be send to http://127.0.0.1/v1/tax_number_validity/bw/1234.

Therefore, we get the following array with split: [http:,127.0.0.1, v1, tax_ number_validity, bw, 1234]

In the Tests the url is different. So, I did not want to change the hardcover values but make it dynamic. You are right, if the len is 0 or smaller than 2, we would get an error. However, we would have also gotten an error with the old values. Furthermore, we should not get a length of 0, because at least tax_number_validity, state_abbreviation and tax_number should be set. Even if they are not set, getting an error (while probably a more explicit one) is fine.

File: acceptance_tests/cypress/integration/lotse_flow.spec.js (hunk 1 above)
Comment (JulianRoesner) [reply acceptance_tests/cypress/integration/lotse_flow.spec.js:218]: As written in the description this test is now covered by the TaxNumberPage.tests. This was only the old way to test the tax number page

File: webapp/tests/elster_client/mock_erica.py (hunk 2 above)
Comment (SannyNguyenHung) [reply webapp/tests/elster_client/mock_erica.py:87]: Okay thank you for the explanation :)
